packages/aws/data_stream/cloudfront_logs/elasticsearch/ingest_pipeline/default.yml

```
@@ -21,51 +21,316 @@ processors:
-        TLS: '(TLS|SSL)'
-        EDGE_LOCATION: '[A-Z]{3}\d+(-[A-Z]+\d+)?'
-        FORWARDED_IPS: '(%{IP}?(,(\s*)%{IP})*)+'
-  - gsub:
-      field: _tmp.time
-      pattern: \t
-      replacement: " "
-      ignore_missing: true
+      separator: "\t"
+      target_fields:
+      - _tmp.date                                   # date
+      - _tmp.time                                   # time
+      - _tmp.x_edge_location                        # x-edge-location
+      - _tmp.sc_bytes                               # sc-bytes
+      - _tmp.c_ip                                   # c-ip
+      - _tmp.cs_method                              # cs-method
+      - _tmp.cs_host                                # cs(Host)
+      - _tmp.cs_uri_stem                            # cs-uri-stem
+      - _tmp.cs_status                              # cs-status
+      - _tmp.cs_referer                             # cs(Referer)
+      - _tmp.cs_user_agent                          # cs(User-Agent)
+      - _tmp.cs_uri_query                           # cs-uri-query
+      - _tmp.cs_cookie                              # cs(Cookie)
+      - _tmp.x_edge_result_type                     # x-edge-result-type
+      - _tmp.x_edge_request_id                      # x-edge-request-id
+      - _tmp.x_host_header                          # x-host-header
+      - _tmp.cs_protocol                            # cs-protocol
+      - _tmp.cs_bytes                               # cs-bytes
+      - _tmp.time_taken                             # time-taken
+      - _tmp.x_forwarded_for                        # x-forwarded-for
+      - _tmp.ssl_protocol                           # ssl-protocol
+      - _tmp.ssl_cipher                             # ssl-cipher
+      - _tmp.x_edge_response_result_type            # x-edge-response-result-type
+      - _tmp.cs_protocol_version                    # cs-protocol-version
+      - _tmp.fle_status                             # fle-status
+      - _tmp.fle_encrypted_fields                   # fle-encrypted-fields
+      - _tmp.c_port                                 # c-port
+      - _tmp.time_to_first_byte                     # time-to-first-byte
+      - _tmp.x_edge_detailed_result_type            # x-edge-detailed-result-type
+      - _tmp.sc_content_type                        # sc-content-type
+      - _tmp.sc_content_len                         # sc-content-len
+      - _tmp.sc_range_start                         # sc-range-start
+      - _tmp.sc_range_end                           # sc-range-end
+
+  ### CSV fields ###################################################################################
+
+  # date
+  # time
+  - script:
+      description: timestamp composed of date and time. Script processor performs better than equivalent set processor.
+      lang: painless
+      source: >-
+        ctx._tmp.timestamp = ctx._tmp.date + 'T' + ctx._tmp.time;
+      if: ctx?._tmp?.date != null && ctx?._tmp?.time != null
```

```suggestion
      if: ctx._tmp?.date != null && ctx._tmp?.time != null
```
(similar change throughout: `ctx` is always non-null)

---

Done